core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -195,6 +198,61 @@ class ControllerApis(val requestChannel: RequestChannel,
       requestThrottleMs => createResponseCallback(requestThrottleMs))
   }
 
+  def handleDeleteTopics(request: RequestChannel.Request): Unit = {
+    if (!config.deleteTopicEnable) {
+      if (request.context.apiVersion() < 3) {
+        throw new InvalidRequestException("Topic deletion is disabled.")
+      } else {
+        throw new TopicDeletionDisabledException()
+      }
+    }
+    val deleteTopicsRequest = request.body[DeleteTopicsRequest]
+    val nameToId = new mutable.HashMap[String, Uuid]
+    deleteTopicsRequest.data().topicNames().iterator().asScala.foreach {
+      name => nameToId.put(name, Uuid.ZERO_UUID)
+    }
+    deleteTopicsRequest.data().topics().iterator().asScala.foreach {
+      nameAndId => nameToId.put(nameAndId.name(), nameAndId.topicId())
+    }
+    val (describable, deletable)  = {
+      if (authHelper.authorize(request.context, DELETE, CLUSTER, CLUSTER_NAME)) {
+        (nameToId.keySet, nameToId.keySet)
+      } else {
+        val authorizedDescribeTopics: Set[String] = authHelper.filterByAuthorized(
```

Comment:
Is it possible the request carries only topic ids? If yes, how to find the related topic names and then authorize them here?

Reply:
thanks, this is a good point.  I fixed the code to accept only topic IDs.  I also found a few more bugs, and added a unit test for the deletion code in controller apis.